spek-runtime/common/src/main/kotlin/org/spekframework/spek2/runtime/lifecycle/LifecycleAwareAdapter.kt

```diff
@@ -9,28 +9,29 @@ import kotlin.reflect.KProperty
 
 sealed class LifecycleAwareAdapter<T>(val factory: () -> T, val destructor: (T) -> Unit)
     : LifecycleAware<T>, LifecycleListener {
-    var cached: T? = null
+    protected data class CachedValue<out T>(val value: T)
```

Comment:
Why protected?

Reply:
Let's make a sealed class instead, so we could have `object` as null value thus we won't need `cached!!.value`

Reply:
* If you mean "why protected instead of private ?", it's because it can't be seen by subclasses if set to private (namely `ScopeCachingModeAdapter`, `TestCachingModeAdapter`, and `GroupCachingModeAdapter`)
* If you mean "why add a visibility modifier ?", then I ask you why should it be world visible ? I don't think that the property is meant to be read or written by other classes. Isn't it. Maybe I'm not understanding correctly the usage of these classes.

OK I'll give it a try with a `sealed class` instead. If I understand correctly it would be something like this ?
```
sealed class CachedValue {
    object Empty : CachedValue()
    data class Value<out T>(val value: T) : CachedValue()
}
```

Reply:
- Ah ok, didn't realize it applies for classes in same file :)
- nono, I was like why not "private" :)
- yep!